community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/logical/Eagerness.scala

```diff
@@ -195,8 +218,13 @@ object Eagerness {
   @tailrec
   def writeReadConflictInHeadRecursive(head: PlannerQuery, tail: PlannerQuery): Boolean = {
     // TODO:H Refactor: This is same as writeReadConflictInTail, but with different overlaps method. Pass as a parameter
-    val conflict = if (tail.queryGraph.writeOnly) false
-    else head.queryGraph writeOnlyHeadOverlaps tail.queryGraph
+    val conflict =
+      if (tail.queryGraph.writeOnly) false
+      else
+        // NOTE: Here we do not check writeOnlyHeadOverlapsHorizon, because we do not know of any case where a
+        // write-only head could cause problems with reads in future horizons
```

Comment:
That's a useful observation. I don't know any way for a write-only head to increase cardinality above one, which makes eagerness analysis trivial.